app/forms/candidate_interface/other_qualification_type_form.rb

```diff
@@ -58,5 +69,15 @@ def intermediate_state
     def qualification_type_changed?
       id && ApplicationQualification.find(id)&.qualification_type != qualification_type
     end
+
+    def prepare(options)
+      options.select { |_, value| value.present? }
+
+      if [A_LEVEL_TYPE, AS_LEVEL_TYPE, GCSE_TYPE].include?(options[:qualification_type])
```

Comment:
This deals with scenarios where a candidate updates the qualification type from a 'non-uk' qualification to a 'uk' qualification